components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/JWTSignatureValidationUtils.java

```diff
@@ -0,0 +1,212 @@
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.validators.jwt.JWKSBasedJWTValidator;
+
+import java.security.PublicKey;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateExpiredException;
+import java.security.cert.CertificateNotYetValidException;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPublicKey;
+
+/**
+ * Utility class for JWT Signature Validation related functionalities.
+ */
+public class JWTSignatureValidationUtils {
+
+    private static final Log log = LogFactory.getLog(JWTSignatureValidationUtils.class);
+
+    private static final String JWKS_URI = "jwksUri";
+    private static final String JWKS_VALIDATION_ENABLE_CONFIG = "JWTValidatorConfigs.Enable";
+    private static final String ENFORCE_CERTIFICATE_VALIDITY
+            = "JWTValidatorConfigs.EnforceCertificateExpiryTimeValidity";
+
+    /**
+     * Method to validate the signature of the JWT.
+     *
+     * @param signedJWT signed JWT whose signature is to be verified.
+     * @param idp       Identity provider who issued the signed JWT
+     * @return whether signature is valid, true if valid else false
+     * @throws JOSEException
+     * @throws IdentityOAuth2Exception
+     */
+    public static boolean validateSignature(SignedJWT signedJWT, IdentityProvider idp)
+            throws JOSEException, IdentityOAuth2Exception {
+
+        boolean isJWKSEnabled = false;
+        boolean hasJWKSUri = false;
+        String jwksUri = null;
+
+        String isJWKSEnalbedProperty = IdentityUtil.getProperty(JWKS_VALIDATION_ENABLE_CONFIG);
+        isJWKSEnabled = Boolean.parseBoolean(isJWKSEnalbedProperty);
+        if (isJWKSEnabled) {
+            if (log.isDebugEnabled()) {
+                log.debug("JWKS based JWT validation enabled.");
+            }
+        }
+
+        IdentityProviderProperty[] identityProviderProperties = idp.getIdpProperties();
+        if (!ArrayUtils.isEmpty(identityProviderProperties)) {
+            for (IdentityProviderProperty identityProviderProperty : identityProviderProperties) {
+                if (StringUtils.equals(identityProviderProperty.getName(), JWKS_URI)) {
+                    hasJWKSUri = true;
+                    jwksUri = identityProviderProperty.getValue();
+                    if (log.isDebugEnabled()) {
+                        log.debug("JWKS endpoint set for the identity provider : " + idp.getIdentityProviderName() +
+                                ", jwks_uri : " + jwksUri);
+                    }
+                    break;
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("JWKS endpoint not specified for the identity provider : " + idp
+                                .getIdentityProviderName());
+                    }
+                }
+            }
+        }
```

Comment:
Would be nice to move this to a private method to keep the main method clean

Reply:
Moved to a private method and removed the `boolean hasJWKSUri`.